extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -0,0 +1,566 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {tryParseJson} from '../../../src/json';
+import {getData} from '../../../src/event-helper';
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {IntersectionObserver} from '../../../src/intersection-observer';
+import {SimplePostMessageApiDef} from '../../../src/simple-postmessage-api-def';
+
+/**
+ * Used to manage messages for different Safeframe ad slots.
+ *
+ * Maps a sentinel value to an instance of the SafeframeHostApi to which that
+ * sentinel value belongs.
+ * @type{!Object<string,SafeframeHostApi>}
```

Comment:
Can we make this !Object<string, !SafeframeHostApi>}, unless we can map to null/undefined values?

Reply:
done